Allow Users to disable SwaggerUI via configuration

gschuurman opened the pull request:
<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->
Due to potential issues we want to give teams the opportunity to disable the swaggerUI. This can now be done via the configuration key: webserver.enable_swagger_ui.
For backwards compatibility this has been set to default true, but for teams willing to disable this it can be set to false.


---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

gschuurman committed:
Allow Users to disable SwaggerUI via configuration

Due to potential issues we want to give teams the opportunity to disable
the swaggerUI. This can now be done via the configuration key:
webserver.enable_swagger_ui.
For backwards compatibility this has been set to default true, but for
teams willing to disable this it can be set to false.

msumit commented on airflow/www/extensions/init_views.py:
```diff
@@ -208,7 +208,11 @@ def _handle_method_not_allowed(ex):
 
     spec_dir = path.join(ROOT_APP_DIR, "api_connexion", "openapi")
-    connexion_app = App(__name__, specification_dir=spec_dir, skip_error_handlers=True)
+    if not conf.getboolean("webserver", "enable_swagger_ui"):
```
If someone updates the code but doesn't update the conf file, this line will raise an error.

gschuurman replied:
This should be fixed in the latest commit. Thanks for noticing

gschuurman committed:
Implement fallback for not updating config file

Users can update the code without updating the config file, this would
have caused an issue. By implementing the fallback we set a default
value. By default the swaggerUI should be enabled

msumit commented on airflow/www/extensions/init_views.py:
```diff
@@ -208,7 +208,11 @@ def _handle_method_not_allowed(ex):
+    if not conf.getboolean("webserver", "enable_swagger_ui", fallback=True):
+        options = {"swagger_ui": False}
+    else:
+        options = {"swagger_ui": True}
```
nit: simplify the code 
`options = {"swagger_ui": conf.getboolean("webserver", "enable_swagger_ui", fallback=True)}`

gschuurman replied:
That's a good suggestion

gschuurman committed:
Simply code using tenary expression

gschuurman committed:
Adjust default configuration based on changes in code

gschuurman committed:
Remove swaggerUI from menu when disabled

Removing the swagger UI via config should also reflect on the ui,
otherwise this change will cause a broken link in the UI

csm10495 committed:
Add a new config for adapting masked secrets to make it easier to prevent secret leakage in logs (#28239)


Co-authored-by: Jarek Potiuk <jarek@potiuk.com>
Co-authored-by: Ash Berlin-Taylor <ash_github@firemirror.com>

uranusjr committed:
Refactor get_mapped_group_summaries for web UI (#28374)

uranusjr committed:
Correctly select a mapped task's "previous" task (#28379)

eladkal committed:
Update README_RELEASE_PROVIDER_PACKAGES.md (#28377)

Make it a bit more friendly for the next one who release for the first time

amoghrajesh committed:
Adding an example dag for dynamic task mapping (#28325)

syedahsn committed:
Add EMR Notebook operators (#28312)

potiuk committed:
Fix failing static checks in main (#28401)

The #28325 implemented some changes to spellcheck scripts via
"isort" - apparently in some cases "docs" is not see as internal
package by isort. This needs to be further investigated but this
PR restores proper sorting.

uranusjr committed:
Add documentation for task group mapping (#28001)

ephraimbuddy commented on airflow/config_templates/config.yml:
```diff
@@ -1502,6 +1502,13 @@
+    - name: enable_swagger_ui
+      description: |
+        Boolean for running SwaggerUI in the webserver.
+      version_added: 2.5.1
```
```suggestion
      version_added: 2.6.0
```
~Should this be in config 'api' section, since swagger is about REST API?~

bbovenzi committed:
Dont show task/run durations when there is no start_date (#28395)

* Dont show broken task/run duration

* add test

TohnJhomas committed:
Restructure Docs  (#27235)

A big restructuring of our docs to make them easier to reason about - in a much more logical and consistent way. Likel other updates and content changes will be needed to make it even better.

Taragolis committed:
Add Amazon Elastic Container Registry (ECR) Hook (#28279)

Co-authored-by: D. Ferruzzi <ferruzzi@amazon.com>
Co-authored-by: Niko <onikolas@amazon.com>

potiuk committed:
fIx isort problems introduced by recent isort release (#28434)

The recent isort changed their mind on sorting the imports. This
change follows the change and bumps isort to latest released
version (isort has no install_requires on its own so bumping
min version has no effect on other dependencies)

This change adds a number of isort:skip_file, isort:off, isort:skips
in order to handle a very annoying bug in isort, that no matter how
much you try, it sometimes treat "known first party" packages
differently - depending on how many files it processes at a time.

We should be able to restore it after this bug is fixed:
https://github.com/PyCQA/isort/issues/2045

This change also updates the common.sql API to skip them from isort
for the very same reason (depending on how many files are modified,
the isort order might change.

potiuk committed:
Fix output buffering for `breeze testing test` (#28433)

Since we disabled warnings and redirected them to the output file,
we lost continuous output of the progress of tests when they
were run as `breeze testing tests` command. The whole summary
of the tests appeared only at the end of tests.

This turned out to be Python output buffering in "filter_out_warning"
class. It is fixed by flushing sys.stdout after every printed line
now.

potiuk committed:
Fix discoverability of tests for ARM in Breeze (#28432)

Breeze in case of ARM processor lacks support for several components
(because they do not have supported ARM binaries available):

* MySQL
* MSSQL
* LevelDB
* Azure Service Bus

When you try to attempt to run pytest on a group of tests that import
one of those, the collection failed and none of the tests could run
even if some of them could.

This change uses pytest's skip on a module level and local imports
in case the tests are inter-mixed with other tests in the same module
to avoid import errors during collection.

The try/except pattern over pytest.importorskip is preferred because
we are using try/except in a number of other cases and we are pretty
familiar with similar pattern and importorskipi has a bit unexpected
behaviour (it returns imported module and you do not see the usual
`import nnnn`. Also in our case we often wrap more than one
import in one try/except (and it would lead to a duplicating messages
to print really.

We also add a separate command in ci to just perform a collection of
tests and see if all tests are collectable after uninstalling all
those libraries. This would prevent the problems from reapparing.

Isort fixes are implemented for recently relesed isort version

potiuk committed:
Re-enable Plyvel on ARM as it now builds cleanly (#28443)

Previously, Plyvel had to be disabled in order to get
ARM compatibility (it did not have
binary wheels released and it failed to compile cleanly on debian).
But the last problem is fixed now, so we can re-enable it for ARM.

IAL32 committed:
Correctly template Glue Jobs `create_job_kwargs` arg (#28403)

0xflotus committed:
fix: very small typo error in readme (#28447)

bbovenzi committed:
Maintain manual scroll position in task logs (#28386)

XD-DENG committed:
Log FileTaskHandler to work with KubernetesExecutor's multi_namespace_mode (#28436)

akakakakakaa committed:
Add `ensure_ascii=False` in trigger dag run API (#28451)

* Add ensure_ascii=False in trigger dag run api

* Fix static checks

gschuurman committed:
Update airflow/config_templates/config.yml

Co-authored-by: Ephraim Anierobi <splendidzigy24@gmail.com>

msumit committed:
Merge branch 'main' into feature/disable_swagger_ui

gschuurman committed:
Solve Static checks
